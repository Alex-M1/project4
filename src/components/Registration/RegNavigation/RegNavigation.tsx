import React from 'react';
import { useTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';
import { useTheme } from 'src/components/hooks/useTheme';
import { StNavLink } from 'src/components/Auth/styled';
import { url } from 'constants/urls';

const RegNavigation = () => {
  const { colors, theme } = useTheme();
  const { t } = useTranslation();
  return (
    <StNavLink
      theme={theme}
      colors={colors}
    >
      <p>{t('to_login')}</p>
      &nbsp;
      <NavLink to={url.authClient}>
        {t('to_login_link')}
      </NavLink>
    </StNavLink>
  );
};

export default RegNavigation;
