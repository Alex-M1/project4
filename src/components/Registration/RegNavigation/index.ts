import RegNavigation from './RegNavigation';

export default RegNavigation;
