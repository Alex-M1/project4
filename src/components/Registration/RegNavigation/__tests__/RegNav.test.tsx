import React from 'react';
import { mountSmart, shallowSmart } from 'tests/testHelper';
import RegNavigation from '../RegNavigation';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
        t: jest.fn(),
    })),
}),
);

describe('RegNavigation', () => {
    let props;
    beforeEach(() => {
        props = {

        };
    });
    it('Should match snapshot', () => {
        const component = shallowSmart(<RegNavigation {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StNavLink', () => {
        const component = mountSmart(<RegNavigation {...props} />);
        expect(component.find('styled__StNavLink')).toHaveLength(1);
    });
    it('should change', () => {
        const component = mountSmart(<RegNavigation {...props} />);
        expect(component.find('styled__StNavLink')).toHaveLength(1);
    });
});
