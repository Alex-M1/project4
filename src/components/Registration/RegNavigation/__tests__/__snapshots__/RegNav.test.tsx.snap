// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegNavigation Should match snapshot 1`] = `"<div class=\\"styled__StNavLink-kYzesN fxjFPh\\"><p></p>Â <a aria-current=\\"page\\" class=\\"active\\" href=\\"/\\"><p class=\\"styled__StP-gOoQqg hCGbld\\"></p></a></div> "`;
